service/src/main/java/bio/terra/workspace/service/resource/controlled/model/ControlledResource.java

```diff
@@ -134,10 +134,13 @@ public abstract void addCreateSteps(
    * to delete the specific cloud resource.
    *
    * @param flight the delete flight
+   * @param userRequest authenticated user
    * @param flightBeanBag bean bag for finding Spring singletons
    */
   public abstract void addDeleteSteps(
-      DeleteControlledResourcesFlight flight, FlightBeanBag flightBeanBag);
+      DeleteControlledResourcesFlight flight,
```

Comment:
I copied the logic from create flow. 
Let me check if its part of beanbag.

Reply:
It looks like input parameters are [a member of the `Flight` base class](https://github.com/DataBiosphere/stairway/blob/develop/stairway/src/main/java/bio/terra/stairway/Flight.java#L16).

Reply:
1. Good point, since even deletion now checks if AWS is enabled via discover environment. Checking
2. as discussed, using userrequest from flight.inputParams

Reply:
(I actually prefer the new behavior if that's the case, as I think it's closer to what I would expect as an admin)

Reply:
Noted, retaining the new behavior and updated the code as per discussion re: userRequest

Reply:
Thank you!